community/cypher/cypher-logical-plans-3.4/src/main/scala/org/neo4j/cypher/internal/v3_4/logical/plans/ProduceResult.scala

```diff
@@ -19,14 +19,19 @@
  */
 package org.neo4j.cypher.internal.v3_4.logical.plans
 
-case class ProduceResult(columns: Seq[String], inner: LogicalPlan) extends LogicalPlan {
-  val lhs = Some(inner)
+import org.neo4j.cypher.internal.ir.v3_4.{CardinalityEstimation, IdName, PlannerQuery, StrictnessMode}
 
-  def solved = inner.solved
-
-  def availableSymbols = inner.availableSymbols
+/**
+  * For every source row, produce a row containing only the variables in 'columns'
```

Comment:
Maybe mention that this is the operator responsible for preparing the results for the end user? In slotted this is where we  move from primitive longs to NodeProxies

Reply:
Hmm, I didn't think about that, because it wasn't reflected in the Pipe. I guess it just happens because of expression evaluation. Now that I did think about it, I'm leaning towards not including that, as the representation of the logical node type could be considered a runtime concern. With that mindframe, it shouldn't be up to the logical plan to dictate how value representations are transformed. WDYT?